### PR TITLE
#218 fixes and improvements

### PR DESCRIPTION
**Pull request by singulart**
Addressed issue #218

### CODE REVIEW

**Review by DzhideX — APPROVED** [review]
If I had to nitpick, maybe 2 things:
- There's a couple of instances inside of if-cases where you could use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to make it a tad bit cleaner and easier to read. (ex. line 69,76)
- And maybe use full variable names where ever possible(like thresh -> threshold)

That being said, I'll leave it up to your discretion if you want to change these as functionality-wise the code looks great! :)

**Comment by singulart** [comment]
@DzhideX Thanks for the prompt review and the feedback. I've addressed your comments. 
@bwhm Can you please merge?

**Comment by DzhideX** [comment]
Sorry about that, I was given write access today but I didn't put 2 and 2 together :)

Merging!

**Pull request merged**
